src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -94,12 +94,14 @@ def __init__(self, mgr: "CephadmOrchestrator"):
     def make_daemon_spec(self, host: str,
                          daemon_id: str,
                          netowrk: str,
-                         spec: ServiceSpecs) -> CephadmDaemonSpec:
+                         spec: ServiceSpecs,
+                         daemon_type: Optional[str] = None,) -> CephadmDaemonSpec:
         return CephadmDaemonSpec(
             host=host,
             daemon_id=daemon_id,
             spec=spec,
-            network=netowrk
+            network=netowrk,
+            daemon_type=daemon_type
```

Comment:
Hm. we should have

```python
assert daemon_id.startswith(daemon_type)
```
at this point.

Reply:
@sebastian-philipp  Can you elaborate on this? I thought daemon names were structured like `daemon_type.daemon_id` in which case daemon_id wouldn't include the daemon type. Am I off on that?